docs/permission/plugin-authors/05-frontend-authorization.md

```diff
@@ -0,0 +1,167 @@
+---
+id: 05-frontend-authorization
+title: 5. Frontend Components with Authorization
+description: Placing frontend components behind authorization
+---
+
+In the previous sections, we learned how to protect our plugin's backend API routes with the permission framework. Most routes that return some data to be displayed (such as our `GET /todos` route) need no additional changes on the frontend, as the backend will simply return an empty list or a `404`. However, for UI elements that trigger a mutative action, it's common practice to hide or disable them when a user doesn't have permission.
+
+Take, for example, the "Add" button in our todo list application. When a user clicks this button, the frontend makes a `POST` request to our backend's `/todos` route. If a user tries to add a todo but is not authorized, they will have no way of knowing this until they perform the action and are faced with an error. This is a poor user experience. We can do better by disabling the add button.
+
+> Note: Placing frontend components behind authorization cannot take the place of placing your backend routes behind authorization. Authorization checks on the frontend should be used in _addition_ to the corresponding backend authorization, as an improvement to the user experience. If you do not place your backend route behind authorization, a malicious actor can still send a request to the route even if you disabled the corresponding frontend component.
+
+## Using `usePermission`
+
+Let's start by adding the packages we will need:
+
+```
+$ yarn workspace @internal/plugin-todo-list \
+  add @backstage/plugin-permission-react @internal/plugin-todo-list-common
+```
+
+Let's make the following changes in `plugins/todo-list/src/components/TodoListPage/TodoListPage.tsx`:
+
+```diff
+...
+  import {
+    alertApiRef,
+    discoveryApiRef,
+    fetchApiRef,
+    useApi,
+  } from '@backstage/core-plugin-api';
++ import { usePermission } from '@backstage/plugin-permission-react';
++ import { todoListCreate } from '@internal/plugin-todo-list-common';
```

Comment:
Probably missed the `*Permission` suffix as part of the example package right? I feel it's very important for readability. What I mean is that this should be `todoListCreatePermission`, and that we should updated ALL the things! accordingly.

Reply:
Done!